src/notification.c

```diff
@@ -516,7 +515,8 @@ int notification_init(notification *n, int id)
                 }
                 printf("skipping notification: %s %s\n", n->body, n->summary);
         } else {
-                g_queue_insert_sorted(queue, n, notification_cmp_data, NULL);
+                if (id == 0 || !notification_replace_by_id(n))
```

Comment:
I don't see how `id` can be 0 at this point, since it's assigned in line 431.

Reply:
keep in mind: `id` is the parameter here and **not the id of the notification**. if `id == 0`, it indicates, that we should not replace a notification.

Reply:
Oh, woops missed that.